crates/stages/src/stages/tx_lookup.rs

```diff
@@ -92,43 +96,60 @@ impl<DB: Database> Stage<DB> for TransactionLookupStage {
             return Ok(ExecOutput::done(input.checkpoint()))
         }
 
-        let (tx_range, block_range, is_final_range) =
-            input.next_block_range_with_transaction_threshold(provider, self.commit_threshold)?;
-        let end_block = *block_range.end();
+        // 500MB temporary files
+        let mut hash_collector: Collector<TxHash, TxNumber> =
+            Collector::new(Arc::new(TempDir::new()?), 500 * (1024 * 1024));
```

Comment:
nit add helper with defaults for tempdir / sizes

Reply:
thinking of increasing size here, maybe 1GB? May make the first part of the stage slightly faster at more memory usage expense

Reply:
this should be configurable to allow lower-end and higher-end hardware operate at their capacity

Reply:
https://github.com/paradigmxyz/reth/issues/6696